imap_processing/codice/codice_l1a.py

```diff
@@ -102,19 +105,89 @@ def _generate_simulated_data(self, length_in_bits):
         random_bytes : bytes
             A list of random bytes to be used as simulated science data
         """
+        print("\tUsing simulated data")
         # Generate string of random bits of proper length
         bit_string = bin(random.getrandbits(length_in_bits))[2:]
         bit_string = bit_string.zfill(length_in_bits)
-        print(f"Length of data in bits: {len(bit_string)}")
+        print(f"\tLength of data in bits: {len(bit_string)}")
 
         # Convert list of random bits into byte-size list of integers
         random_bytes_str = [bit_string[i : i + 8] for i in range(0, len(bit_string), 8)]
         random_bytes_int = [int(item, 2) for item in random_bytes_str]
         random_bytes = bytes(random_bytes_int)
-        print(f"Length of data in bytes: {len(random_bytes)}")
+        print(f"\tLength of data in bytes: {len(random_bytes)}")
 
         return random_bytes
 
+    def create_science_dataset(self):
+        """Create an ``xarray`` dataset for the unpacked science data.
+
+        The dataset can then be written to a CDF file.
+
+        Returns
+        -------
+        xarray.Dataset
+            xarray dataset containing the science data and supporting metadata
+
+        # TODO: Pull out common code and put in codice.utils alongside
+        # create_hskp_dataset()
+        """
+        # Temporary workaround to get epoch in correct dimensions
+        epoch_times = [launch_time for index in range(len(self.data))]
+
+        # Define the attributes specific to species data
+        species_attrs = dataclasses.replace(
+            cdf_attrs.l1a_science_attrs,
+            depend_0="species",
+            depend_1=None,
+            depend_2=None,
+            catdesc="TBD",
+            fieldname="TBD",
+            label_axis="TBD",
+        ).output()
+
+        epoch_time = xr.DataArray(
+            epoch_times,
+            name="epoch",
+            dims=["epoch"],
+            attrs=ConstantCoordinates.EPOCH,
+        )
+        species_array = xr.DataArray(
+            name="species",
+            data=np.zeros(36864),
+            dims=("species"),
+            attrs=species_attrs,
+        )
+
+        dataset = xr.Dataset(
+            coords={"epoch": epoch_time, "species": species_array},
+            attrs=cdf_attrs.codice_l1a_global_attrs.output(),
+        )
+
+        # Create a data variable for each species
+        for species_data, species_name in zip(self.data, LO_SW_SPECIES_NAMES):
+            species_data_int = [byte for byte in species_data]
```

Comment:
Can you put `list(species_data)` or just `species_data` directly into your `DataArray` constructor below?

Reply:
Thanks, I didn't piece together that `list(species_data)` does the same thing. I just wanted a list of integers from the byte data.